src/main/test/com/epam/esm/dao/GiftCertificateDaoTest.java

```diff
@@ -0,0 +1,107 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.config.SpringConfig;
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.mapper.GiftCertificateMapper;
+import com.epam.esm.utils.QueryConstructor;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ContextConfiguration(classes = SpringConfig.class)
+@ActiveProfiles("dev")
+public class GiftCertificateDaoTest {
+
+    private QueryConstructor constructor;
+    private GiftCertificateMapper mapper;
+    private GiftCertificateDao dao;
+    private GiftCertificate giftCertificate;
+    @Autowired
+    private JdbcTemplate template;
+
+    @Before
+    public void init() {
+        constructor = new QueryConstructor();
+        mapper = new GiftCertificateMapper();
+        giftCertificate = new GiftCertificate();
+        dao = new GiftCertificateDao(template, mapper, constructor);
```

Comment:
You shouldn't create all these objects like this using `new` keyword. As we're using Spring we need to make use of it in test as well and create som test Spring context and inject beans.